appcues/src/main/java/com/appcues/ui/modal/DialogModal.kt

```diff
@@ -33,48 +43,119 @@ private val MAX_WIDTH_EXPANDED_DP = 560.dp
 private val MAX_HEIGHT_COMPACT_DP = Dp.Unspecified
 private val MAX_HEIGHT_MEDIUM_DP = 800.dp
 private val MAX_HEIGHT_EXPANDED_DP = 900.dp
-private const val SCREEN_PADDING = 0.05
+
+private val SCREEN_HORIZONTAL_PADDING = 12.dp
+private val SCREEN_VERTICAL_PADDING = 24.dp
+
+internal data class DialogTransition(
+    val type: String,
+    val slideInEdge: String?,
+    val slideOutEdge: String?,
+)
+
+internal enum class TransitionEdge {
+    LEADING, TRAILING, TOP, BOTTOM, CENTER
+}
 
 @Composable
 internal fun DialogModal(
     style: ComponentStyle?,
     content: @Composable (modifier: Modifier, containerPadding: PaddingValues, safeAreaInsets: PaddingValues) -> Unit,
-    windowInfo: AppcuesWindowInfo
+    windowInfo: AppcuesWindowInfo,
+    transition: DialogTransition,
 ) {
-    val configuration = LocalConfiguration.current
-    val dialogHorizontalMargin = (configuration.screenWidthDp * SCREEN_PADDING).dp
-    val dialogVerticalMargin = (configuration.screenHeightDp * SCREEN_PADDING).dp
     val isDark = isSystemInDarkTheme()
+    val density = LocalDensity.current
 
     val maxWidth = maxWidthDerivedOf(windowInfo)
     val maxHeight = maxHeightDerivedOf(windowInfo)
 
-    AppcuesContentAnimatedVisibility(
-        enter = dialogEnterTransition(),
-        exit = dialogExitTransition(),
+    val horizontalPaddingPx = with(density) { SCREEN_HORIZONTAL_PADDING.roundToPx() }
+    val verticalPaddingPx = with(density) { SCREEN_VERTICAL_PADDING.roundToPx() }
+
+    Box(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(horizontal = SCREEN_HORIZONTAL_PADDING, vertical = SCREEN_VERTICAL_PADDING)
```

Comment:
also an important note - the padding is being applied here, at the root of the composition, outside of the animation block, rather than inside on the surface as before (L63 in older code). It seemed like this was required to give the animation the proper area to transition things. It might have also been possible before to have a very large shadow clipped off by the container box from what I was seeing? Need to double check that, but I think this should be safe.

Reply:
in any case you need Box to be outside the AnimatedVisibility